src/analysis/normed_space/lattice_ordered_group.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2021 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import analysis.normed_space.basic
+import algebra.lattice_ordered_group
+import topology.order.lattice
+
+/-!
+# Normed lattice ordered groups
+
+Motivated by the theory of Banach Lattices, we then define `normed_lattice_add_comm_group` as a
+lattice with a covariant normed group addition satisfying the solid axiom.
+
+## Main statements
+
+We show that a normed lattice ordered group is a topological lattice with respect to the norm
+topology.
+
+## References
+
+* [Meyer-Nieberg, Banach lattices][MeyerNieberg1991]
+
+## Tags
+
+normed, lattice, ordered, group
+-/
+
+/-!
+### Normed lattice orderd groups
+
+Motivated by the theory of Banach Lattices, this section introduces normed lattice ordered groups.
+-/
+
+local notation `|`a`|` := abs a
+
+/--
+Let `α` be a normed commutative group equipped with a partial order covariant with addition, with
+respect which `α` forms a lattice. Suppose that `α` is *solid*, that is to say, for `a` and `b` in
+`α`, with absolute values `|a|` and `|b|` respectively, `|a| ≤ |b|` implies `∥a∥ ≤ ∥b∥`. Then `α` is
+said to be a normed lattice ordered group.
+-/
+class normed_lattice_add_comm_group (α : Type*)
+  extends normed_group α, lattice α :=
+(add_le_add_left : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b)
+(solid : ∀ a b : α, |a| ≤ |b| → ∥a∥ ≤ ∥b∥)
+
+lemma solid {α : Type*} [normed_lattice_add_comm_group α] {a b : α} (h : |a| ≤ |b|) : ∥a∥ ≤ ∥b∥ :=
+  normed_lattice_add_comm_group.solid a b h
+
+/--
+A normed lattice ordered group is an ordered additive commutative group
+-/
+@[priority 100] -- see Note [lower instance priority]
+instance normed_lattice_add_comm_group_to_ordered_add_comm_group {α : Type*}
+  [h : normed_lattice_add_comm_group α] : ordered_add_comm_group α := { ..h }
+
+/--
+A normed lattice ordered group is a lattice ordered group
+-/
+@[priority 100] -- see Note [lower instance priority]
+instance normed_lattice_add_comm_group_to_covariant_class {α : Type*}
+  [h : normed_lattice_add_comm_group α] : covariant_class α α (+) (≤) :=
+{elim := λ a b c bc,  normed_lattice_add_comm_group.add_le_add_left _ _ bc a}
+
+/--
+Let `α` be a normed group with a partial order. Then the order dual is also a normed group.
+-/
+@[priority 100] -- see Note [lower instance priority]
+instance {α : Type*} : Π [normed_group α], normed_group (order_dual α) := id
+
+lemma test {α : Type*} [h: normed_lattice_add_comm_group α] (a b : α) : a ≤ b ↔ -b ≤ -a :=
+begin
+  exact neg_le_neg_iff.symm,
+end
+
```

Comment:
Remove. (Personally, I would not complain if you turned this into an `example`, but I understand the consensus is not to have lemmas like this.)

Reply:
Sorry, this was left in by mistake. Removed.